sim/play.py

```diff
@@ -219,15 +262,55 @@ def play(args: argparse.Namespace) -> None:
             if num_episodes > 0:
                 env_logger.log_rewards(infos["episode"], num_episodes)
 
+        if args.log_krec:
+            frame = krec.KRecFrame(
+                video_timestamp=start_time_ns + int(t * env.dt * 1e9),  # Convert simulation time to real time
+                frame_number=t,
+                inference_step=t // env_cfg.control.decimation,
+            )
+
+            # Add actuator states and commands for each joint
+            for i in range(env.num_dof):
+                state = krec.ActuatorState(
+                    actuator_id=i,
+                    online=True,
+                    position=env.dof_pos[robot_index, i].item(),
+                    velocity=env.dof_vel[robot_index, i].item(),
+                    torque=env.torques[robot_index, i].item(),
+                )
+                command = krec.ActuatorCommand(
+                    i,  # actuator id
+                    position=actions[robot_index, i].item(),
+                    velocity=0.0,  # if you have velocity commands
+                    torque=actions[robot_index, i].item(),
+                )
+                frame.add_actuator_state(state)
+                frame.add_actuator_command(command)
+
+            # Add IMU data
+            imu_values = krec.IMUValues(
+                gyro=krec.Vec3(x=obs[0, 0], y=obs[0, 1], z=obs[0, 2]),
+                quaternion=krec.IMUQuaternion(x=obs[0, 3], y=obs[0, 4], z=obs[0, 5], w=obs[0, 6]),
+            )
+            frame.set_imu_values(imu_values)
+            krec_logger.add_frame(frame)
+
     env_logger.print_rewards()
-    env_logger.plot_states()
+    # env_logger.plot_states()
```

Comment:
remove or flag?

Reply:
commented because it also throws a (non-crashing error), so I think we should remove it unless its important. 

```
Traceback (most recent call last):
  File "/opt/anaconda3/envs/ksl/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/opt/anaconda3/envs/ksl/lib/python3.8/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/home/kasm-user/ali_repos/sim/sim/utils/logger.py", line 71, in _plot
    fig, axs = plt.subplots(nb_rows, nb_cols)
NameError: name 'plt' is not defined
```

Reply:
Can you fix it or remove if the dependencies will cause issues.

Reply:
TLDR from debugging: Plotting fails due to some multiprocessing. Need to either:
Remove multiprocessing and plot in main thread
Fix data sharing between processes
Use different visualization approach

Reply:
kk